loki/transform/transform_utilities.py

```diff
@@ -27,10 +27,96 @@
 
 __all__ = [
     'convert_to_lower_case', 'replace_intrinsics', 'sanitise_imports',
-    'replace_selected_kind'
+    'replace_selected_kind', 'single_variable_declarations', 'single_variable_declaration'
 ]
 
 
+def single_variable_declaration(routine, variables=()):
+    """
+    Modify/extend variable declarations to declare variables specified in ``variables`` in single declarations.
+
+    Parameters
+    ----------
+    routine: :any:`Subroutine`
+        The subroutine in which to modify the variable declarations
+    variables: tuple
+        Variables to grant unique/single declaration for
+    """
+    if variables:
+        decl_map = {}
+        for decl in FindNodes(VariableDeclaration).visit(routine.spec):
+            if len(decl.symbols) > 1:
+                convert = False
+                symbols = []
+                unique_symbols = []
+                for smbl in decl.symbols:
+                    if smbl.name in variables:
+                        convert = True
+                        unique_symbols.append(smbl)
+                    else:
+                        symbols.append(smbl)
+                if convert:
+                    counter = 1
+                    if symbols:
+                        decl_map[decl] = decl.clone(symbols=as_tuple(symbols))
+                    else:
+                        decl_map[decl] = None
+                    for smbl in unique_symbols:
+                        routine.spec.insert(routine.spec.body.index(decl) + counter, (decl.clone(symbols=(smbl,)),))
+                        counter += 1
+        routine.spec = Transformer(decl_map).visit(routine.spec)
+
+
+def single_variable_declarations(routine, strict=True):
+    """
+    Modify/extend variable declarations to only declare one variable each time while preserving the order.
+
+    Parameters
+    ----------
+    routine: :any:`Subroutine`
+        The subroutine in which to modify the variable declarations
+    strict: bool
+        Whether to strictly make unique variable declarations or to only disassemble non-arrays and arrays and among
+        arrays, arrays with differing shapes.
+    """
+    decl_map = {}
+    for decl in FindNodes(VariableDeclaration).visit(routine.spec):
+        if len(decl.symbols) > 1:
+            if strict:
+                counter = 1
+                for i_sdecl, sdecl in enumerate(decl.symbols):
```

Comment:
I think this could be a single list comprehension, similar to above:
```python
decl_map[decl] = tuple(decl.clone(symbols=(s,)) for s in decl.symbols)
```